networkx/linalg/modularitymatrix.py

```diff
@@ -0,0 +1,75 @@
+"""Modularity matrix of graphs.
+"""
+#    Copyright (C) 2004-2015 by
+#    Aric Hagberg <hagberg@lanl.gov>
+#    Dan Schult <dschult@colgate.edu>
+#    Pieter Swart <swart@lanl.gov>
+#    All rights reserved.
+#    BSD license.
+from __future__ import division
+import networkx as nx
+from networkx.utils import not_implemented_for
+
+__author__ = "\n".join(['Aric Hagberg <aric.hagberg@gmail.com>',
+                        'Pieter Swart (swart@lanl.gov)',
+                        'Dan Schult (dschult@colgate.edu)',
+                        'Jean-Gabriel Young <jean.gabriel.young@gmail.com>'])
+__all__ = ['modularity_matrix']
+
+
+@not_implemented_for('directed')
+@not_implemented_for('multigraph')
+def modularity_matrix(G, nodelist=None):
+    """Return the modularity matrix of G.
+
+    The modularity matrix is the matrix B = A - <A>, where A is the adjacency
+    matrix and <A> is the average adjacency matrix, assuming that the graph
+    is descibred by the configuration model.
+
+    Parameters
+    ----------
+    G : graph
+       A NetworkX graph
+
+    nodelist : list, optional
+       The rows and columns are ordered according to the nodes in nodelist.
+       If nodelist is None, then the ordering is produced by G.nodes().
+
+    Returns
+    -------
+    B : Numpy matrix
+      The modularity matrix of G.
+
+    Notes
+    -----
+    For MultiGraph/MultiDiGraph, the edges weights are summed.
```

Comment:
Oops, it slipped through (I used `linealg/laplacianmatrix.py` as a base template)
My apologies.